src/main/java/ui/MenuControl.java

```diff
@@ -89,47 +87,71 @@ private Menu createFileMenu() {
     }
 
     /**
-     * Logs the user action and creates a new board
+     * Checks whether the current board is dirty, i.e. has unsaved changes.
+     *
+     * @return true if current board is dirty, false otherwise
      */
-    private void onBoardNew() {
-        logger.info("Menu: Boards > New");
-        newBoard();
+    public final boolean isCurrentBoardDirty() {
+        if (!prefs.getLastOpenBoardPanelInfos().isPresent()) {
+            return true;
+        }
+        return !panels.getCurrentPanelInfos().equals(prefs.getLastOpenBoardPanelInfos().get());
     }
 
     /**
-     * Creates an empty board. User will be prompted to
-     * confirm the action if there are unclosed panels
+     * Prompts a dialog to ask the user whether or not to save the current board.
      */
-    public final void newBoard() {
-        if (!isNewBoardCreationDialogConfirmed()) {
-            logger.info("New board creation cancelled");
+    public final void promptToSaveCurrentBoard() {
+        boolean shouldSave = DialogMessage.showYesNoConfirmationDialog("Save Changes?",
+                "All unsaved changes will be discarded.",
+                "Do you want to save them?",
+                "Yes", "No");
+
+        if (!shouldSave) {
+            logger.info("User abandoned unsaved changes.");
             return;
         }
 
-        panels.closeAllPanels();
-        saveBoardAs();
+        saveBoard();
+        logger.info("Changes to the current board saved.");
     }
 
     /**
-     * Returns a boolean indicating whether we can proceed with
-     * the new board creation. If there are no panels open, it
-     * returns true; else a dialog is shown for the user to confirm.
+     * Prompts a {@link BoardNameDialog} to ask the user for a new board name
+     * @return the user's response
      */
-    private boolean isNewBoardCreationDialogConfirmed() {
-        List<PanelInfo> panelList = panels.getCurrentPanelInfos();
-        if (panelList.isEmpty()) {
-            return true;
+    public final Optional<String> promptForBoardName() {
+        BoardNameDialog dlg = new BoardNameDialog(prefs, mainStage);
+        Optional<String> response = dlg.showAndWait();
+
+        return response;
+    }
+
+    /**
+     * Logs the user action and creates a new board
```

Comment:
Remove `Logs the user action and...`. Logging is irrelevant to method description.